### ISSUE TITLE
ConfigureSmartEnum not working

### ISSUE THREAD

**Issue by HAOYI0425** [ardalis/SmartEnum#554]
**this is my ConfigureConventions**
![image](https://github.com/user-attachments/assets/d3fd71b8-19b2-498f-817d-0817f33d02d8)

**The smart enum**
![image](https://github.com/user-attachments/assets/578f6d39-fe22-4291-99c3-deb0db758e64)

**How I use it under entity**
![image](https://github.com/user-attachments/assets/ba4f4c5c-831f-4b75-b46f-706518b8fc2a)

`The exception 'No suitable constructor was found for entity type 'UnitMaterial'. The following constructors had parameters that could
not be bound to properties of the entity type: 
    Cannot bind 'name', 'value' in 'UnitMaterial(string name, int value)'`
    
how can I fix this ? please help
UnitMaterial and Entity is under my DOMAIN project, and dbContext is under INFRASTRUCTURE project

My specs
.NET 8.0
SmartEnum 8.1
SmartEnum.EFCore 8.1

**Comment by HAOYI99** [comment]
can someone please help on this

**Comment by ardalis** [comment]
It looks right as is - if you create a minimal repo I can take a look.

As a workaround you can try this:

```csharp
protected override void OnModelCreating(ModelBuilder modelBuilder)
{
    modelBuilder.Entity<YourEntity>()
        .Property(e => e.Unit)
        .HasConversion(
            v => v.Value, // Convert `SmartEnum` to its `value` (int)
            v => UnitMaterial.FromValue(v) // Convert `value` (int) back to `SmartEnum`
        );
}
```

**Comment by HAOYI99** [comment]
> It looks right as is - if you create a minimal repo I can take a look.
> 
> As a workaround you can try this:
> 
> ```cs
> protected override void OnModelCreating(ModelBuilder modelBuilder)
> {
>     modelBuilder.Entity<YourEntity>()
>         .Property(e => e.Unit)
>         .HasConversion(
>             v => v.Value, // Convert `SmartEnum` to its `value` (int)
>             v => UnitMaterial.FromValue(v) // Convert `value` (int) back to `SmartEnum`
>         );
> }
> ```

Hi @ardalis, thanks for the reply.

yes, I did use this method to manual map the conversion for each SmartEnum.

However, the [document](https://github.com/ardalis/SmartEnum?tab=readme-ov-file#using-smartenumefcore) mentioned ConfigureSmartEnum is sufficient for configuring all SmartEnum.

I tried many times, ends up I create a SmartEnumConverter myself. 
```
public class SmartEnumConverter<TEnum>() : ValueConverter<TEnum, string>(
    enumValue => enumValue.Name,
    stringValue => SmartEnum<TEnum, int>.FromName(stringValue, true)
) where TEnum : SmartEnum<TEnum, int>;
```

and the `builder.ConfigureSmartEnum();` is not required with this approach
```
protected override void ConfigureConventions(ModelConfigurationBuilder builder)
{
    base.ConfigureConventions(builder);
    builder.Properties<DateOnly>().HaveConversion<DateOnlyConverter>().HaveColumnType("date");
    builder.Properties<DateTime>().HaveColumnType("datetime");
    builder.Properties<TimeSpan>().HaveConversion<long>();
    builder.Properties<Enum>().HaveConversion<string>();
    builder.ConfigureSmartEnum<UnitMaterial>();
    builder.ConfigureSmartEnum<AmTechnologies>();
}
```

**Comment by bengavin** [comment]
We are running into this same error when generating migrations since moving to .NET 9.  Any entity that has changed since the previous migration and contains a smart enum generates this exception during the migration generation process, unless the workaround given by @ardalis is used in the entity configuration.

**Comment by ardalis** [comment]
Yeah EF Core must have changed something in v9 I guess. We'll take a look and figure out how to do a more general solution (again) as soon as we can. Community assistance appreciated as always, including reproductions of the problem in clean repos so if we come up with a fix we can test it with your scenario.

**Comment by jczacharia** [comment]
I encountered a similar issue:
```cs
    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);
        modelBuilder.ConfigureSmartEnum(); // Won't work here
        modelBuilder.ApplyConfigurationsFromAssembly(typeof(DbCtx).Assembly);
    }
```
I had to add it after `ApplyConfigurationsFromAssembly`, which seemed to do the trick.
```cs
    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);
        modelBuilder.ApplyConfigurationsFromAssembly(typeof(DbCtx).Assembly).ConfigureSmartEnum(); // Works here
    }
```
